### ISSUE TITLE
Not sure if PlaceholderBindings::insert() needs a check of allocZero() similar to allocate()

### ISSUE THREAD

**Issue by dinkdeep** [pytorch/glow#4266]
Looks like PlaceholderBindings::insert () in  lib/Graph/PlaceholderBindings.cpp  does not seems to have the check for tensor to be conditionally zeroed out.
if (P->allocZero()) {
    T->zero();
  }
Similar to Tensor *PlaceholderBindings::allocate(Placeholder *P)
Not sure if its needed.

**Comment by jfix71** [comment]
@dinkdeep Thanks for raising this. I think the way it currently is setup is correct (i.e. we should not zero the tensor in `insert()`) -- the doxygen of `Placeholder::allocZero()` says `\returns True if associated Tensors should be zeroed when allocated.` However `PlaceholderBindings::insert()` does not allocate a tensor -- it simply adds it to the bindings mapped to the Placeholder.

**Comment by dinkdeep** [comment]
Thanks alot Jordan My bad, I think you are right.

The misunderstanding occurred when I was trying to run through one of the C++ Sample App test case TEST_P(MLTest, convNetForImageRecognition) {) in MLTest.cpp . 

I noticed some strange/seemingly large  values [4321339579369492789915025408.00] in the outputs of the filters of convolution occurring due to the gradient updated by the backdrop , and misinterpreted that they could be due to uninitialised values which were getting added up. 

But latter realised that the convolutions are being initialised "InitKind::Xavier" by  Graph.cpp  
bindings.allocate(filter)->init(glow::Tensor::InitKind::Xavier, fanIn, getPRNG()); 
which is a requirement for the  random initialization of weights to get a reasonable minimum during training .

The final filter output was carrying these types of values(7492186235932490238587317728051]) which motivated me to conclude that there could be some uninitialised values .

BTW I added a dump in libjit Matmul and found that the values in the filter associated were like this below and so were in the final filter out .

I haven't read through the math of Xavier  in initXavier() but will definitely go through it and cross check it out and may be run some unit test cases of Xavier.

fwdElemMul - filter__1_GsumMult__1
LHS filter__1_gsum
shape: ( 16 5 5 1 )
max: 1580216856672284221721967260439  min: 0.000
[[[[7492186235932490238587317728051],
[0.000],
[0.000],
[1128648319523968115086983168.00],
[17466494273175158784.000]],

[[16468725962671503966208.000],
[18690969741873429224620032.000],
[4321339579369492789915025408.00],
[17751091086526403462159138816.0],
[17110709902612889600.000]],

[[0.000],
[0.000],
[67336524591750973489152.000],
[73169484375904654525502324736.0],
[17406461847181288982708224.000]],
